h2o-algos/src/main/java/hex/tree/gbm/GBM.java

```diff
@@ -916,6 +917,21 @@ private void rollupMinMaxPreds(DTree tree, DTree.Node node, float[] mins, int mi
       return new GBMModel(modelKey, parms, new GBMModel.GBMOutput(GBM.this));
     }
 
+    @Override
+    protected void doInTrainingCheckpoint() {
+      try {
+        String modelFile = _parms._in_training_checkpoints_dir + "/" + _model._key.toString() + "." + _model._output._ntrees;
+        Key keybackup = _model._key;
+        _model.setInputParms(_parms);
+        _model._key = Key.make(_model._key + "." +  _model._output._ntrees);
+        _model._output.changeModelMetricsKey(_model._key);
+        _model.exportBinaryModel(modelFile, true);
+        _model._key = keybackup;
+        _model._output.changeModelMetricsKey(_model._key);
```

Comment:
is it too expensive to clone the model instead of fooling around with this modify-save-restore logic?

Reply:
Yes, it is, and not expensive, thanks!